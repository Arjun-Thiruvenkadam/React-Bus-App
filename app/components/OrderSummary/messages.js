/*
 * OrderSummary Messages
 *
 * This contains all the text for the OrderSummary component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.OrderSummary';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Order Summary',
  },
  headMessage: {
    id: `${scope}.headMessage`,
    defaultMessage: 'Enjoy a safe and comfortable ride',
  },
  tickets: {
    id: `${scope}.tickets`,
    defaultMessage: 'Your tickets',
  },
  checkout: {
    id: `${scope}.checkout`,
    defaultMessage: 'Continue to checkout?',
  },
  cancel: {
    id: `${scope}.cancel`,
    defaultMessage: 'Cancel',
  },
  continue: {
    id: `${scope}.continue`,
    defaultMessage: 'Continue',
  },
});
