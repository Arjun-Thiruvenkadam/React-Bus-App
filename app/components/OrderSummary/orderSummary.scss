.orderSummary {
  p {
    display: inline-block;
    border: 3px solid white;
    margin: auto;
  }
  h2 {
    color: #0f3460;
  }
}
