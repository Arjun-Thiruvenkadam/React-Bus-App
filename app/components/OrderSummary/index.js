/**
 *
 * OrderSummary
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import Button from '../Button/index';
import './orderSummary.scss';

const OrderSummary = props => {
  const ticketsId = props.tickets.sort((a, b) => a.ticketId - b.ticketId);
  const tickets = ticketsId.map(ticket => (
    <p key={ticket.ticketId}>{ticket.ticketId} </p>
  ));
  return (
    <div className="orderSummary">
      <h2>
        <FormattedMessage {...messages.header} />
      </h2>
      <h4>
        <FormattedMessage {...messages.headMessage} />
      </h4>
      <h4>
        <FormattedMessage {...messages.tickets} />
      </h4>
      {tickets}
      <h4>
        <FormattedMessage {...messages.checkout} />
      </h4>
      <Button clicked={props.cancelOrder} btnType="Danger">
        <FormattedMessage {...messages.cancel} />
      </Button>
      <Button clicked={props.completeOrder} btnType="Success">
        <FormattedMessage {...messages.continue} />
      </Button>
    </div>
  );
};

OrderSummary.propTypes = {
  completeOrder: PropTypes.func,
  cancelOrder: PropTypes.func,
  tickets: PropTypes.any,
};

export default OrderSummary;
