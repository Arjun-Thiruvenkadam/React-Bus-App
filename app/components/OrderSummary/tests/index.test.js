import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import OrderSummary from '../index';

describe('<OrderSummary />', () => {
  it('should render and match the snapshot', () => {
    const tickets = [
      {
        _id: '5f435a78fe45e2e4d6c35143',
        ticketId: 1,
        status: 'open',
        personId: null,
        __v: 0,
      },
      {
        _id: '5f435a78fe45e2e4d6c35144',
        ticketId: 2,
        status: 'open',
        personId: null,
        __v: 0,
      },
    ];
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <OrderSummary tickets={tickets} />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
