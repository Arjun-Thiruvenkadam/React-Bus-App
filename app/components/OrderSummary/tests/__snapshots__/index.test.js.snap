// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderSummary /> should render and match the snapshot 1`] = `
<div
  class="orderSummary"
>
  <h2>
    <span>
      Order Summary
    </span>
  </h2>
  <h4>
    <span>
      Enjoy a safe and comfortable ride
    </span>
  </h4>
  <h4>
    <span>
      Your tickets
    </span>
  </h4>
  <p>
    1
     
  </p>
  <p>
    2
     
  </p>
  <h4>
    <span>
      Continue to checkout?
    </span>
  </h4>
  <button
    class="Button Danger"
  >
    <span>
      Cancel
    </span>
  </button>
  <button
    class="Button Success"
  >
    <span>
      Continue
    </span>
  </button>
</div>
`;
